go.mod

```diff
@@ -4,13 +4,17 @@ go 1.21
 
 require (
 	github.com/confluentinc/confluent-kafka-go v1.9.2
+	github.com/fxamacker/cbor/v2 v2.5.0
 	github.com/google/uuid v1.3.1
 	github.com/grpc-ecosystem/go-grpc-middleware/providers/prometheus v1.0.0
 	github.com/hashicorp/go-hclog v1.5.0
 	github.com/hashicorp/go-plugin v1.5.2
 	github.com/jpillora/backoff v1.0.0
 	github.com/linkedin/goavro/v2 v2.12.0
+	github.com/mitchellh/mapstructure v1.5.0
 	github.com/mwitkow/grpc-proxy v0.0.0-20230212185441-f345521cb9c9
+	github.com/onsi/ginkgo v1.14.1
+	github.com/onsi/gomega v1.29.0
```

Comment:
We started moving away from these recently for various reasons. Are you using any particular features of these libraries? If not, pure go tests would be preferred. (especially in this repo, where we are trying to keep dependencies light)

Reply:
Ack, I can see if there's another common way to do these kinds of tests.  Just used to doing it that way.

Reply:
Sounds good. I think the high level structure maps to regular subtests well, and we usually lean on testify require/assert.